robotiq_force_torque_sensor/nodes/rq_sensor.cpp

```diff
@@ -91,11 +133,20 @@ static void decode_message_and_do(INT_8 const  * const buff, INT_8 * const ret)
 bool receiverCallback(robotiq_force_torque_sensor::sensor_accessor::Request& req,
 	robotiq_force_torque_sensor::sensor_accessor::Response& res)
 {
-	ROS_INFO("I heard: [%s]",req.command.c_str());
-	INT_8 buffer[512];
-	decode_message_and_do((char*)req.command.c_str(), buffer);
-	res.res = buffer;
-	ROS_INFO("I send: [%s]", res.res.c_str());
+	/// Support for old string-based interface
+	if (req.command.length())
+	{
+		ROS_WARN("Usage of command-string is deprecated, please use the numeric command_id");
+		ROS_INFO("I heard: [%s]",req.command.c_str());
+		INT_8 buffer[512];
+		decode_message_and_do((char*)req.command.c_str(), buffer);
+		res.res = buffer;
+		ROS_INFO("I send: [%s]", res.res.c_str());
+		return true;
+	}
+
+	/// New interface with numerical commands
+	decode_message_and_do(req, res);
```

Comment:
Oh...I see.  You call it here.  Why not return the result of this method call (add a return bool to it).  The `decode_message_and_do` method could fail (i.e. res.success = false), but this code always returns true.